app/controllers/oauth2_controller.rb

```diff
@@ -76,6 +78,18 @@ def authenticate_with_edx(username, auth_code)
       head 502
     end
 
+    def authenticate_with_github(auth_state, auth_code)
+      oauth2_error('no_authorization_code') && return unless auth_code.present?
+      oauth2_error('no_auth_state') && return unless auth_state.present?
+
+      auth = GithubAuthenticator.new(auth_state, auth_code)
+      login = auth.authenticate!
+
+      render json: { access_token: login.oauth2_token, github_token: auth.github_token }
```

Comment:
Wanted ability to utilize the user's actual github oauth token for queries against github's api on their behalf ... maybe a better way to achieve this?

Reply:
I believe the OAuth 2.0 spec doesn’t allow to respond with any custom attributes. After all, github would just be used as an identity provider here anyway. That doesn’t mean github‘s ath Code couldn’t be persisted and exposed to the client but I think this is not the right endpoint and it’s probably out of scope for rails_api_auth.